talisker/gunicorn.py

```diff
@@ -221,24 +278,8 @@ def init(self, parser, opts, args):
         cfg['pre_request'] = gunicorn_pre_request
 
         if pkg_is_installed('prometheus-client'):
-            # only available in gunicorn 19.7+
-            if hasattr(gunicorn.config, 'ChildExit'):
-                # we can do the full cleanup
-                cleanup = talisker.metrics.prometheus_cleanup_worker
-                server_hook = 'child_exit'
-            else:
-                # weaksauce cleanup
-                from prometheus_client import multiprocess
-                cleanup = multiprocess.mark_process_dead
-                server_hook = 'worker_exit'
-
-            def prometheus_multiprocess_worker_exit(server, worker):
-                "Worker cleanup function for multiprocess prometheus_client."
-                logging.getLogger(__name__).info(
-                    'Performing multiprocess prometheus_client cleanup')
-                cleanup(worker.pid)
-
-            cfg[server_hook] = prometheus_multiprocess_worker_exit
+            cfg['on_starting'] = gunicorn_on_starting
+            cfg['child_exit'] = gunicorn_child_exit
```

Comment:
This is the default behaviour in talisker, but can still be completely overriden by users via standard gunicorn config, which is the right thing to do to me. Am I right?

Reply:
Yes, that's right. If you override too many things yourself though stuff may break.